core/src/main/java/tech/tablesaw/joining/DataFrameJoiner.java

```diff
@@ -26,19 +27,32 @@
 import tech.tablesaw.selection.BitmapBackedSelection;
 import tech.tablesaw.selection.Selection;
 
+import java.util.Arrays;
+import java.util.List;
 import java.util.concurrent.atomic.AtomicInteger;
 
 public class DataFrameJoiner {
 
     private static final String TABLE_ALIAS = "T";
 
     private final Table table;
-    private final CategoricalColumn<?> column;
+    private CategoricalColumn<?>[] columns;
+    private String[] columnNames;
     private AtomicInteger joinTableId = new AtomicInteger(2);
 
-    public DataFrameJoiner(Table table, String column) {
+    /**
+     * Constructor.
+     * @param table       The table to join on
+     * @param columnNames The column names to join on
+     */
+    public DataFrameJoiner(Table table, String... columnNames) {
         this.table = table;
-        this.column = table.categoricalColumn(column);
+        columns = new CategoricalColumn<?>[columnNames.length];
+        this.columnNames = columnNames;
+        for(int i = 0; i < this.columnNames.length; i++) {
```

Comment:
add a space after `for`

Reply:
Could simplify this as `for (String colName : columnNames)`

Reply:
ok, no need then. I didn't see that `i` was being used anywhere else

Reply:
You're right - it's not being used anywhere else in that immediate method - but later on in the processing in another method the logic relies on the same index variable coordinating the name to the column.  This seemed like the simplest/cleanest solution.  Thanks.  I'll prepare and submit a new update with fixes for the other changes soon.